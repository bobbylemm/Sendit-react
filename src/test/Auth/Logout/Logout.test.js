import { mount, shallow } from 'enzyme';
import React from 'react';
import LogOut, { Logout, mapDispatchToProps } from '../../../Auth/Logout';

const props = {
  history: { push: jest.fn() },
  logoutUsers: jest.fn()
};

describe('Test the snapshot of the logout page', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<LogOut {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should test mapDispatchToProps', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).logoutUsers();
    expect(dispatch.mock.calls[0][0]).toBeDefined();
  });
  it('should test the mounted component', () => {
    const wrapper = mount(<Logout {...props} />);
    expect(props.history.push).toHaveBeenLastCalledWith('/');
    expect(wrapper).toEqual({});
  });
});
