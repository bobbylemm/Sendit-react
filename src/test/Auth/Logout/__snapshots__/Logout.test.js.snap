// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the snapshot of the logout page should match the snapshot 1`] = `ShallowWrapper {}`;
